src/main/java/com/commercetools/sync/types/utils/TypeUpdateFieldDefinitionActionUtils.java

```diff
@@ -0,0 +1,277 @@
+package com.commercetools.sync.types.utils;
+
+import com.commercetools.sync.commons.exceptions.BuildUpdateActionException;
+import com.commercetools.sync.commons.exceptions.DuplicateKeyException;
+import com.commercetools.sync.commons.exceptions.DuplicateNameException;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.types.FieldDefinition;
+import io.sphere.sdk.types.Type;
+import io.sphere.sdk.types.commands.updateactions.AddFieldDefinition;
+import io.sphere.sdk.types.commands.updateactions.ChangeFieldDefinitionOrder;
+import io.sphere.sdk.types.commands.updateactions.RemoveFieldDefinition;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static com.commercetools.sync.commons.utils.CommonTypeUpdateActionUtils.buildUpdateAction;
+import static com.commercetools.sync.types.utils.FieldDefinitionUpdateActionUtils.buildActions;
+import static java.lang.String.format;
+import static java.util.Collections.singletonList;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+
+public final class TypeUpdateFieldDefinitionActionUtils {
```

Comment:
Weird naming, I also noticed the weird naming in the ProductType sync :)

I would suggest to name it: `FieldDefinitionsUpdateActionUtils`

Reply:
https://github.com/commercetools/commercetools-sync-java/commit/6f1484541de42487b69083d4556201bc2cb40080#diff-e071adde6fef713b55f6b5e2f7e683faL31